lexos/views/content_analysis_view.py

```diff
@@ -0,0 +1,213 @@
+        if 'dictionary_labels' in session:
+            dict_labels = session['dictionary_labels']
+        else:
+            dict_labels = []
+        if 'active_dictionaries' in session:
+            active_dicts = session['active_dictionaries']
+        else:
+            active_dicts = [True] * len(dict_labels)
+        if 'toggle_all_value' in session:
+            toggle_all_value = session['toggle_all_value']
+        else:
+            toggle_all_value = True
+        if 'formula' in session:
+            formula = session['formula']
+        else:
+            formula = ""
+        return render_template('contentanalysis.html',
+                               dictionary_labels=dict_labels,
+                               active_dictionaries=active_dicts,
+                               toggle_all_value=toggle_all_value,
+                               formula=formula)
+    else:
+        num_active_docs = detect_active_docs()
+        active_dicts = ContentAnalysisReceiver().options_from_front_end(
+        ).active_dicts
+        dict_labels = ContentAnalysisReceiver().options_from_front_end(
+        ).dict_labels
+        session['formula'] = ContentAnalysisReceiver().options_from_front_end(
+        ).formula
+        if len(dict_labels) == 0:
+            dict_labels = [os.path.splitext(dict_name)[0]
+                           for dict_name in dictionary_names]
+            active_dicts = [True] * len(dict_labels)
+        num_active_dicts = active_dicts.count(True)
+        if num_active_docs == 0 and num_active_dicts == 0:
+            return error("At least 1 active document and 1 active "
+                         "dictionary are required to perform a "
+                         "content analysis.")
+        elif num_active_docs == 0:
+            return error("At least 1 active document is required to perform "
+                         "a content analysis.")
+        elif num_active_dicts == 0:
+            return error("At least 1 active dictionary is required to perform"
+                         " a content analysis.")
+        file_manager = load_file_manager()
+        active_files = file_manager.get_active_files()
+        for file in active_files:
+            analysis.add_file(file_name=file.name,
+                              label=file.label,
+                              content=file.load_contents())
+        for dict_name, dict_label, active in zip(dictionary_names,
+                                                 dict_labels,
+                                                 active_dicts):
+            if active:
+                f = open(os.path.join(path, dict_name), "r")
+                content = f.read()
+                analysis.add_dictionary(file_name=dict_name,
+                                        label=dict_label,
+                                        content=content)
+        result_table, formula_errors = analysis.analyze()
+        if len(formula_errors) != 0 or result_table is None:
```

Comment:
Formula error can be None, according to your documentation. Calling len function on None can generate error.

Reply:
I'm sue formula_errors is always a string, and clicking analyze without setting the formula should cause an error if formula_error was set to None, but it doesen't.

Reply:
I just fixed the documentation